torch2trt/torch2trt.py

```diff
@@ -865,3 +892,164 @@ def set_layer_precision(ctx, layer):
         layer.precision = trt.float16
         layer.set_output_type(0, trt.float16)
 
+        
+
+# from torch2trt.torch2trt import (
+#     torch2trt, 
+#     trt,
+#     tensorrt_converter,
+#     get_conversion_context,
+#     get_arg
+# )
+
+
+# SHAPE WRAPPING
+_int = int
+_tuple = tuple
+_int_mul = int.__mul__
+_int_add = int.__add__
+_int_sub = int.__sub__
+_int_floordiv = int.__floordiv__
+
+class IntWrapper(int):
+    
+    @property
+    def _trt(self):
+        if not hasattr(self, '_raw_trt'):
+            ctx = get_conversion_context()
+            self._raw_trt = ctx.network._network.add_constant([1], np.array([_int(self)], dtype=np.int32)).get_output(0)
```

Comment:
I've been wondering about how this works... Normally, we would detach the value here when creating a constant TRT value, but in this case it looks like we're just passing the wrapper directly into the (numpy array) constant layer instead? Since we use this class primarily for shape values, when the shapes change, does this TRT value change because the underlying int value is changing?

Reply:
This adds the _raw_trt value if it doesn't already exists.

However, _raw_trt is set 

1. When the integer is created as part of a tensor shape (see _shape_wrapper line 1041 below)
2. When the integer is the result of arithmetic of other integers (see operations below)

Reply:
Ah okay after stepping through the code some more, I think I understand how this progression works now.. 
I understood how the `_raw_trt` was getting set before, but was confused as to how this was tracking the shape changes as the model was being fed differently shaped inputs. (In the updated PR) it turns out this was just getting set here:
```
      shape_trt = ctx.network._network.add_shape(input._trt).get_output(0)

      for i, d in enumerate(output):
          d._raw_trt = ctx.network._network.add_slice(shape_trt, [i], [1], [1]).get_output(0)
```
Specifically, we just set the `_raw_trt` for each dimension to be the output of the shape layer, which will correctly track the shape of the tensor regardless of the input tensor shape.